_posts/2022-04-11-1.15-announcement.md

```diff
@@ -0,0 +1,402 @@
+amounts of data which could lead to huge operator states. This is why we introduced watermark alignment
+in this release.
+
+For sources based on the new source interface, 
+[watermark alignment](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/dev/datastream/event-time/generating_watermarks/#watermark-alignment-_beta_)
+can be activated. Users can define 
+alignment groups to pause consuming from sources which are too far ahead from others. The ideal scenario for aligned watermarks is when there are two or more 
+sources that produce watermarks at a different speed and when the source has the same 
+parallelism as splits/shards/partitions.
+
+
+### SQL version upgrades
+
+The execution plan of SQL queries and its resulting topology is based on optimization 
+rules and a cost model. This means that even minimal changes could introduce a completely 
+different topology. This dynamism makes guaranteeing snapshot compatibility 
+very challenging across different Flink versions. In the efforts of 1.15, the community focused 
+on keeping the same query (via the same topology) up and running even after upgrades. 
+
+At the core of SQL upgrades are JSON plans 
+([please note that we only have documentation in our JavaDocs for now and are still working on updating the documentation](https://nightlies.apache.org/flink/flink-docs-release-1.15/api/java/org/apache/flink/table/api/CompiledPlan.html)), which are JSON functions that make it easier to import and export structured data in SQL. This has been introduced for 
+internal use already in previous releases and will now be exposed externally. Both the Table API 
+and SQL will provide a way to compile and execute a plan which guarantees the same 
+topology throughout different versions. This feature will be released as an experimental MVP. 
+Users who want to give it a try already can create a JSON plan that can then be used 
+to restore a Flink job based on the old operator structure. The full feature can be expected 
+in Flink 1.16.
+
+Reliable upgrades makes Flink SQL more dependable for production use cases in the long term. 
+
+
+### Changelog state backend
+
+In Flink 1.15, we introduced the MVP feature of the
+[changelog state](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/ops/state/state_backends/#enabling-changelog)
+backend, which aims at 
+making checkpoint intervals shorter with the following advantages:
+
+1. Less work on recovery: The more frequently a checkpoint is taken, the fewer events 
+   need to be re-processed after recovery.
+2. Lower latency for transactional sinks: Transactional sinks commit on checkpoints, 
+   so faster checkpoints mean more frequent commits.
+3. More predictable checkpoint intervals: Currently the length of the checkpoint mainly
+   depends on the size of the artifacts that need to be persisted on the checkpoint 
+   storage and by keeping the size small, it becomes more predictable.
```

Comment:
1. Shorter end-to-end latency: end-to-end latency mostly depends on the checkpointing mechanism, especially for transactional sinks. Transactional sinks commit on checkpoints, so faster checkpoints mean more frequent commits.
2. More predictable checkpoint intervals: Currently checkpointing time largely depends on the size of the artifacts that need to be persisted on the checkpoint storage. By keeping the size consistently small, checkpointing time becomes more predictable.
3. Less work during recovery: With more frequently checkpoints are taken, less data need to be re-processed after each recovery.